### PR TITLE
Chain source errors

### PR DESCRIPTION
**Pull request by hadley**
To make it easier to understand where problems are coming from

### CODE REVIEW

File: R/source.R [hunk 1]
```diff
@@ -36,7 +36,15 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
   if (wrap) {
     invisible(test_code(NULL, exprs, env))
   } else {
-    invisible(eval(exprs, env))
+    tryCatch(
```
Comment (lionel-) [inline R/source.R:39]: To allow recover and deep backtraces:

```suggestion
    withCallingHandlers(
```

File: R/source.R [hunk 2]
```diff
@@ -36,7 +36,15 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
   if (wrap) {
     invisible(test_code(NULL, exprs, env))
   } else {
-    invisible(eval(exprs, env))
+    withCallingHandlers(
+      invisible(eval(exprs, env)),
+      error = function(err) {
+        abort(
+          paste0("Failed to source ", encodeString(path, quote = '"')),
```
Comment (lionel-) [inline R/source.R:43]:
```suggestion
          paste0("Can't source ", encodeString(path, quote = '"')),
```

Comment (hadley) [reply R/source.R:43]: I wonder if I like "failed" when the problem is not directly related to the argument supplied.

But maybe this should actually just be `In: `

Comment (lionel-) [reply R/source.R:43]: oh yeah looks like a good candidate for an info "In" bullet

**Pull request merged**